### PR TITLE
Stabilize loss calculations by clamping KL divergence values

### PR DESCRIPTION
**Pull request by syo093c**
### Summary

This PR improves the numerical stability of PPO training in `verl` by clamping KL divergence-related values in the loss calculations. Specifically:

- In `compute_policy_loss`, the `negative_approx_kl` value is now clamped to the range \([-10, 10]\) before exponentiation and further use.
- In `kl_penalty` (for the `"low_var_kl"` mode), the KL value is also clamped to \([-10, 10]\) before further calculations.

### Motivation

During PPO training, extreme log-probability differences can occasionally occur, leading to numerical instabilities or exploding/vanishing gradients. By clamping these values, we ensure more stable and reliable training dynamics, especially in edge cases.

### Changes

- Added `torch.clamp` to `negative_approx_kl` in `compute_policy_loss`.
- Added `torch.clamp` to KL values in `kl_penalty` for `"low_var_kl"` mode.
- Both are clamped to the range \([-10, 10]\).

### Related Issues

#891 #721

### CODE REVIEW

File: verl/trainer/ppo/core_algos.py [hunk 1]
```diff
@@ -446,6 +446,8 @@ def compute_policy_loss(
     assert clip_ratio_c > 1.0, "The lower bound of the clip_ratio_c for dual-clip PPO should be greater than 1.0," + f" but get the value: {clip_ratio_c}."
 
     negative_approx_kl = log_prob - old_log_prob
+    # Clamp negative_approx_kl for stability
+    negative_approx_kl = torch.clamp(negative_approx_kl, min=-10.0, max=10.0)
```
Comment (vermouth1992) [inline verl/trainer/ppo/core_algos.py]: Is [-10, 10] too strict for log probs?

Comment (vermouth1992) [reply verl/trainer/ppo/core_algos.py]: Maybe we want to choose the maximum value such as exp is not nan?

Comment (syo093c) [reply verl/trainer/ppo/core_algos.py]: Hi, thank you for your comment!

Float32 max value is about ~3.4e38.
So, we can use `80` as limit (exp(80)=5.5406e+34).
But, choose `80` will still have risk to overflow pg_loss`= advantages*ratio`.

Consideration of numerical stability and exploration space,
I think maybe at most `20` or `30` is enough.

```
exp(20)=4.8517e+08
exp(30)=1.0686e+13
```

Comment (vermouth1992) [reply verl/trainer/ppo/core_algos.py]: Could you make the bound to be [-20, 20] and let's merge this

Comment (syo093c) [reply verl/trainer/ppo/core_algos.py]: Thank you for your advice!
I have set the bound to [-20,20]!

**Comment by syo093c** [comment]
@vermouth1992 

Hi, apologies for reaching out during your busy schedule. I’ve adjusted the range to [-20, 20]. Could we proceed with merging this PR? I’d really appreciate it!

I also noticed that some tests were canceled. Is there anything I can assist with to help resolve this? Thank you so much!

**Comment by hiyouga** [comment]
@vermouth1992 I think we can merge it, WDYT?

**Comment by vermouth1992** [comment]
> @vermouth1992 I think we can merge it, WDYT?

I didn't have a time to verify it. I guess it's better to find a case that will actually cause Nan and verify this fix. In my training experiences, most nan occurs due to machine problems...

**Comment by vermouth1992** [comment]
> > @vermouth1992 I think we can merge it, WDYT?
> 
> I didn't have a time to verify it. I guess it's better to find a case that will actually cause Nan and verify this fix. In my training experiences, most nan occurs due to machine problems...

Oh, I saw a case from EasyR1. Then, we can merge it!

**Review by vermouth1992 — APPROVED** [review]
